cmd/gateway-api/main.go

```diff
@@ -17,9 +46,123 @@ func main() {
 }
 
 func Initialize() {
+	var err error
+	kubeCluster := false
 	utils.AviLog.Infof("AKO is running with version: %s", version)
-	for {
-		time.Sleep(8 * time.Second)
-		utils.AviLog.Infof("AKO is processing the objects...")
+	// Check if we are running inside kubernetes. Hence try authenticating with service token
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		utils.AviLog.Warnf("We are not running inside kubernetes cluster. %s", err.Error())
+	} else {
+		utils.AviLog.Info("We are running inside kubernetes cluster. Won't use kubeconfig files.")
+		kubeCluster = true
+	}
+	if !kubeCluster {
+		cfg, err = clientcmd.BuildConfigFromFlags(masterURL, kubeconfig)
+		utils.AviLog.Infof("master: %s", masterURL)
+		if err != nil {
+			utils.AviLog.Fatalf("Error building kubeconfig: %s", err.Error())
+		}
+	}
+
+	akoControlConfig := gwlib.AKOControlConfig()
+	var gwApiClient *gwApi.Clientset
+
+	gwApiClient, err = gwApi.NewForConfig(cfg)
+	if err != nil {
+		utils.AviLog.Fatalf("Error building gateway-api clientset: %s", err.Error())
+	}
+	akoControlConfig.SetGatewayAPIClientset(gwApiClient)
+
+	//dynamicClient, err := lib.NewDynamicClientSet(cfg)
+	//if err != nil {
+	//	utils.AviLog.Warnf("Error while creating dynamic client %v", err)
+	//}
```

Comment:
we can remove this since the gateway pod won't use this.

Reply:
removed comments